### ISSUE TITLE
GeolocateControl stops tracking user location after user has moved the map

### ISSUE THREAD

**Issue by Doff3n** [visgl/react-map-gl#1445]
#### Description

In our map implementation we would like to:
1. Locate and fly to the user location when the user opens the map 
2. The user then uses the map and puts the phone away. 
3. After a while the user comes back and opens the map and we want to fly to the user location again, same as 1. 

Using a combination of 'auto' and 'trackUserLocation: true' using the GeolocateControl component we are able to implement step 1 in our wanted functionality. But this introduces a functionality where locating the user is disabled when the user moves/pans the map in these lines: https://github.com/visgl/react-map-gl/blob/db08714945b106062a9e8b87722a8c0e343eb3de/src/components/geolocate-control.js#L92

A workaround for us is to toggle the 'auto' prop between true and false with react state and set the showUserLocation to true, resulting in the props:  'auto: true|false' and 'showUserLocation: true' dropping the trackUserLocation prop. But we then loose functionality as this disables the ability for the user to enable and disable the tracking of the location. We also loose the animation on the Geolocate icon when the location is getting a GPS fix.

It seems logical to only set 'trackUserLocation: true', but another issue on this repo has already introduced the auto prop to fix this: https://github.com/visgl/react-map-gl/issues/1191

#### Repro Steps
Create an InteractiveMap with a GeolocateControl component, set the props to: 'trackUserLocation: true' and 'auto'

``` html
<GeolocateControl
    positionOptions={{ enableHighAccuracy: true }}
    trackUserLocation={true}
    auto
    onViewportChange={(newViewport: any) => {
      setViewport({ ...newViewport });
    }}
/>
```

Video of the issue:
https://user-images.githubusercontent.com/611783/116096337-80788d00-a6a9-11eb-9618-bbe29706a349.mov

The same functionality is implemented here: https://visgl.github.io/react-map-gl/examples/locate-user

#### Environment:

- Library Version: react-map-gl 6.1.12
- Mapbox Version: 2.1.1
- React Version: 16.2.0
- Browser Version: Chrome 90.0.4430.85 
- OS: Mac OS X 11.2.3

**Comment by Pessimistress** [comment]
Looks to me that the control is doing what they are supposed to do; since it has no way to know that the user "puts the phone away and comes back", your application has to restore its tracking state somehow.

Would it work if we add a `trigger()` method to the component just like the native control?

**Comment by Doff3n** [comment]
Yes exactly a trigger to enable the location tracking again would be nice. 

The problem for us is that the location tracking is disabled in our implementation in 1. and I found no way of triggering it again with code.

**Comment by Pessimistress** [comment]
This is implemented in v7.0.
